File: jiho/level1/없는숫자더하기.java
package level1;

public class 없는숫자더하기 {
    public static void main(String[] args) {
        Solution86051 solution = new Solution86051();
        int[] numbers = {1, 2, 3, 4, 6, 7, 8, 0};
        int result = solution.solution(numbers);
        System.out.println(result);
    }
}

class Solution86051 {
    public int solution(int[] numbers) {
        int answer = 45;
        for(int number : numbers){
            answer -= number;
        }
        return answer;
    }
}

File: ningpop/level1/없는_숫자_더하기.py
# 2022.09.11
# 풀이 시간: 4분 39초
# 채점 결과: 정답
# 시간복잡도: O(N)
# 문제 링크: https://school.programmers.co.kr/learn/courses/30/lessons/86051

def solution(numbers: list) -> int:
    table = [0] * 10
    for i in numbers:
        table[i] += 1
    answer = 0
    for i in range(len(table)):
        if table[i] == 0:
            answer += i
    return answer

print(solution([1,2,3,4,6,7,8,0])) # 14
print(solution([5,8,4,0,6,7,9])) # 6